### PR TITLE
Parallelism improvement

### PR DESCRIPTION
**Pull request by murgeyseb**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Code improvement.

**What is the current behavior?** *(You can also link to an open issue here)*
Currently, parallel computation on pre processors implies some memory issues that can occur on big cases, with many contingencies.

**What is the new behavior (if this is a feature change)?**
The improvement is to limit the number of parallel network variant used during parallel computation management.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

### CODE REVIEW

File: ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                LOGGER.error("Cannot offer variant '{}' in pool. Should not happen", variantId);
+                throw new AssertionError(String.format("Cannot offer variant '{}' in pool. Should not happen", variantId));
+            }
+        }
+        network.getVariantManager().allowVariantMultiThreadAccess(true);
+    }
+
+    public String getAvailableVariant() throws InterruptedException {
+        LOGGER.info("Searching for available variant for computation");
+        String polledVariant = variantsQueue.take();
```
Comment (TheMaskedTurtle) [inline ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: Rename availableVariant

File: ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        this.network = network;
+        this.initialVariant = initialVariant;
+        this.variantsQueue = new ArrayBlockingQueue<>(numberOfAvailableVariants);
+        initAvailableVariants(numberOfAvailableVariants);
+    }
+
+    private void initAvailableVariants(int numberOfAvailableVariants) {
+        for (int i = 0; i < numberOfAvailableVariants; i++) {
+            String variantId = initialVariant + " variant modified " + i;
+            LOGGER.info("Initializing variants pool with variant '{}'", variantId);
```
Comment (TheMaskedTurtle) [inline ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: Filling ?

File: ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            if (!isSuccess) {
+                LOGGER.error("Cannot offer variant '{}' in pool. Should not happen", variantId);
+                throw new AssertionError(String.format("Cannot offer variant '{}' in pool. Should not happen", variantId));
+            }
+        }
+        network.getVariantManager().allowVariantMultiThreadAccess(true);
+    }
+
+    public String getAvailableVariant() throws InterruptedException {
+        LOGGER.info("Searching for available variant for computation");
```
Comment (TheMaskedTurtle) [inline ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: "Searching available variant for computation"

File: ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        String polledVariant = variantsQueue.take();
+        network.getVariantManager().cloneVariant(initialVariant, polledVariant, true);
+        LOGGER.info("Return variant '{}'", polledVariant);
```
Comment (TheMaskedTurtle) [inline ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: "Variant '{}' will be used" ?

File: ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.*;
+
+/**
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ */
+public class FaraoVariantsPool {
```
Comment (TheMaskedTurtle) [inline ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: Add a delete method which cleans the created network variants ?

Comment (murgeyseb) [reply ra-optimisation/closed-optimisation-rao/src/main/java/com/farao_community/farao/closed_optimisation_rao/pre_processors/FaraoVariantsPool.java]: done

**Pull request merged**